### DISCUSSION TITLE
Is that possible to use non-linear diffusivity of electrodes in the Many-Particle Model?

### DISCUSSION THREAD

**Discussion by meckwell** [pybamm-team/PyBaMM#2066] [Q&A] [answered]
Dear PyBaMM Team,

I wondered whether the MPM in PyBaMM can use the non-linear diffusivity of electrodes in the simulations. I know MPM can use constant diffusivity of electrodes as shown in Toby's paper.

Many thanks for your help in advance.

BW,

Han Yuan (RA Imperial College London)

**Comment by valentinsulzer** [comment, marked as answer]
Yes, it can. Just give diffusivity as a function of concentration and it will work.

**Reply by meckwell** [reply]
Thank you Tino for the prompt answer. A quick follow-up, I notice the Double-Particle model mentioned in one of Toby's papers. I was wondering whether the Double-Particle model is available in PyBaMM now.  It seems to be as efficient as MPM in some cases but the complexity is reduced.

**Reply by valentinsulzer** [reply]
Is this just the same as doing a MPM with only 2 grid points in the size dimension @tobykirk ?

**Reply by tobykirk** [reply]
Yes, although the MPM was not explicitly designed for that, picking 2 grid points in the size dimension gives exactly the double-particle model in that paper. If you use the finite volume discretization and uniform grid spacing in the size domain then, e.g. with `R_min=0` and `R_max=2' and 2 grid points (or volumes), the 2 particles sizes are the centres of the volumes, so 0.5 and 1.5 (and by choosing `R_min, R_max` you can make these whatever you want).

Also, the size distribution `f_a(R)` is specified as a function currently, but you can just input a piecewise constant function that takes the value you want at each of the two grid points.

**Reply by meckwell** [reply]
Toby and Tino, Thank you for the anwswers!
